share/sql/sql.go

```diff
@@ -423,169 +338,141 @@ func (m *mgr) isProjectAdmin(u *userpb.User, path string) bool {
 
-	query, params, err := m.appendUidOwnerFilters(ctx, query, params)
-	if err != nil {
-		return nil, err
-	}
+	query := m.db.Model(&model.Share{}).
+		Where("uid_owner = ? or uid_initiator = ?", uid, uid).
+		Where("orphan = ?", false)
 
-	rows, err := m.db.Query(query, params...)
-	if err != nil {
-		return nil, err
+	// Append filters
+	m.appendFiltersToQuery(query, filters)
+
+	var shares []model.Share
+	var cs3shares []*collaboration.Share
+	res := query.Find(&shares)
+	if res.Error != nil {
+		return nil, res.Error
 	}
-	defer rows.Close()
 
-	var s conversions.DBShare
-	shares := []*collaboration.Share{}
-	for rows.Next() {
-		if err := rows.Scan(&s.UIDOwner, &s.UIDInitiator, &s.ShareWith, &s.Prefix, &s.ItemSource, &s.ItemType, &s.ID, &s.STime, &s.Permissions, &s.ShareType); err != nil {
-			continue
-		}
-		gtype, _ := m.getUserType(ctx, s.ShareWith)
-		// if err != nil {
-		// failed to resolve grantee's user type, TODO Log
-		// }
-		shares = append(shares, conversions.ConvertToCS3Share(s, gtype))
+	for _, s := range shares {
+		granteeType, _ := m.getUserType(ctx, s.ShareWith)
+		cs3share := s.AsCS3Share(granteeType)
+		cs3shares = append(cs3shares, cs3share)
 	}
-	if err = rows.Err(); err != nil {
-		return nil, err
+
+	if len(shares) > 0 {
+		fmt.Printf("First cs3share has id %s\n", cs3shares[0].Id.OpaqueId)
 	}
 
-	return shares, nil
+	return cs3shares, nil
 }
 
 // we list the shares that are targeted to the user in context or to the user groups.
 func (m *mgr) ListReceivedShares(ctx context.Context, filters []*collaboration.Filter) ([]*collaboration.ReceivedShare, error) {
 	user := appctx.ContextMustGetUser(ctx)
-	uid := conversions.FormatUserID(user.Id)
 
-	params := []interface{}{uid, uid, uid, uid}
-	for _, v := range user.Groups {
-		params = append(params, v)
-	}
+	query := m.db.Model(&model.Share{}).
+		Where("orphan = ?", false)
 
-	query := `SELECT coalesce(uid_owner, '') as uid_owner, coalesce(uid_initiator, '') as uid_initiator, lower(coalesce(share_with, '')) as share_with,
-	            coalesce(fileid_prefix, '') as fileid_prefix, coalesce(item_source, '') as item_source, coalesce(item_type, '') as item_type,
-				ts.id, stime, permissions, share_type, coalesce(tr.state, 0) as state
-			  FROM oc_share ts LEFT JOIN oc_share_status tr ON (ts.id = tr.id AND tr.recipient = ?)
-			  WHERE (orphan = 0 or orphan IS NULL) AND (uid_owner != ? AND uid_initiator != ?)`
-	if len(user.Groups) > 0 {
-		query += " AND ((lower(share_with)=lower(?) AND share_type = 0) OR (share_type = 1 AND lower(share_with) in (?" + strings.Repeat(",?", len(user.Groups)-1) + ")))"
-	} else {
-		query += " AND (lower(share_with)=lower(?) AND share_type = 0)"
+	// Also search by all the groups the user is a member of
+	innerQuery := m.db.Where("share_with = ? and shared_with_is_group = ?", user.Username, false)
+	for _, group := range user.Groups {
+		innerQuery = innerQuery.Or("share_with = ? and shared_with_is_group = ?", group, true)
```

Comment:
Can we try this query for the user with most groups to validate it works?

Reply:
This will result in ~ the same query as we do right now, which I remember Giuseppe saying will indeed give issues. 
I discussed earlier with Giuseppe and he was a fan of combining this refactor with the "group membership table" where we can then do just a join

Reply:
I would leave that for a second iteration. We merge this PR and work on that next.

Reply:
Yes, I agree, as discussed with both of you